posthog/batch_exports/service.py

```diff
@@ -930,3 +926,37 @@ def fetch_earliest_backfill_start_at(
         return min(results)
     else:
         raise ValueError(f"Invalid model: {model}")
+
+
+@dataclass(kw_only=True)
+class BatchExportInsertInputs:
+    """Base dataclass for batch export insert inputs containing common fields."""
+
+    team_id: int
+    data_interval_start: str | None
+    data_interval_end: str
+    exclude_events: list[str] | None = None
+    include_events: list[str] | None = None
+    run_id: str | None = None
+    # TODO: Remove after updating existing batch exports to use backfill_details
+    is_backfill: bool = False
+    backfill_details: BackfillDetails | None = None
+    batch_export_model: BatchExportModel | None = None
+    # TODO: Remove after updating existing batch exports
+    batch_export_schema: BatchExportSchema | None = None
```

Comment:
Just a small nit, let's put all the "to delete" stuff at the bottom: 
```suggestion
    backfill_details: BackfillDetails | None = None
    batch_export_model: BatchExportModel | None = None
    # TODO: Remove after updating existing batch exports
    batch_export_schema: BatchExportSchema | None = None
    # TODO: Remove after updating existing batch exports to use backfill_details
    is_backfill: bool = False
```